### PR TITLE
fix(git): scan linked worktrees by admin gitdir so a deleted checkout can't brick the secret scan

### PR DESCRIPTION
**Pull request by devxoul**
## Problem

The canonical-secret Git-history guard (`src/git/secret-history.ts`) enumerated linked worktrees by their **working-directory path** and probed each with `git -C <path>`. When a worktree's working directory was deleted out from under Git — e.g. a subagent creates a worktree under `/tmp/<branch>` and `/tmp` later reaps the directory — the worktree stays registered as `prunable`, `git -C <gone path>` fails with `fatal: cannot change to '<path>': No such file or directory`, and the `runGit` helper turns that into a `GitSecretHistoryError`. Because the guard fails **closed** on any git error, the entire scan aborts and **all model-driven git/bash (and `gh`) is disabled** until an operator manually prunes the stale worktree.

This is a real incident: an agent's PR-review run was bricked mid-review by a leftover `/tmp` worktree, with the only remedy being manual `git worktree prune` + restart.

## Fix

Enumerate linked worktrees by their **administrative gitdir** (`$GIT_COMMON_DIR/worktrees/<id>`) and probe them via `--git-dir=<admin>` instead of `git -C <working-dir>`. The admin gitdir survives a deleted working directory, so the same root-ref / pseudoref / index inspection works for **live and prunable worktrees alike**, and a reaped `/tmp` checkout can no longer break the scan.

Simply dropping a prunable worktree from the scan would be **unsafe** — a linked worktree's own metadata is a genuine concealment surface:

- Its per-worktree pseudorefs (`HEAD` / `FETCH_HEAD` / `MERGE_HEAD`) can pin a **non-commit** (e.g. a tag peeling to a tree carrying `secrets.json`) that `fsck --unreachable` / `rev-list` deliberately ignore — the pseudoref is the only signal that makes that tree a blocking live root.
- Its **retained index** can keep a staged canonical secret reachable while remaining invisible to the main worktree's index and to `fsck`.

So both are now scanned through the admin gitdir: per-worktree root refs (via the existing `--include-root-refs` + fallback path) and the per-worktree index (`ls-files --cached`).

Enumeration itself **fails closed**: only a genuinely absent `worktrees/` directory (`ENOENT`) is treated as "no linked worktrees"; a permission/IO error or an unexpected non-directory entry under `worktrees/` raises `GitSecretHistoryError` rather than silently skipping every per-worktree ref store and index.

## Tests

Added regressions (each Red without the fix, Green with it):

1. Prunable worktree with a commit-only `HEAD` whose working dir was deleted → **allowed** (no false brick — the original bug).
2. Prunable worktree whose admin `FETCH_HEAD` → tag → tree carries `secrets.json` → **blocked** (`unattributable dangling Git objects`).
3. `secrets.json` staged in a prunable worktree's retained index → **blocked** (`["secrets.json"]`).
4. Unreadable `worktrees/` directory → **fails closed**.
5. Unexpected non-directory entry under `worktrees/` → **fails closed**.

## Verification

- `bun run typecheck` ✅
- `bun run lint` ✅ (0 errors)
- `bun run format:check` ✅
- `bun run test` ✅ (11625 pass, 0 fail)

Design reviewed with the Oracle consultant across two rounds; this fix incorporates its findings on both the concealment-surface and fail-open-enumeration edge cases.

### CODE REVIEW

File: src/git/secret-history.ts [hunk 1]
```diff
@@ -132,11 +182,11 @@ async function scanNonCommitRefRoots(agentDir: string, gitArgs: readonly string[
 async function collectRootRefOids(
   agentDir: string,
   gitArgs: readonly string[],
-  linkedWorktrees: readonly string[],
+  linkedAdminDirs: readonly string[],
 ): Promise<string[]> {
   const probes: { cwd: string; args: readonly string[] }[] = [
     { cwd: agentDir, args: gitArgs },
-    ...linkedWorktrees.map((path) => ({ cwd: path, args: [] as readonly string[] })),
+    ...linkedAdminDirs.map((adminDir) => ({ cwd: agentDir, args: [`--git-dir=${adminDir}`] as readonly string[] })),
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/git/secret-history.ts]: Good catch — confirmed and fixed. The helper (now `collectOtherWorktreeGitDirs`) resolves both `--git-dir` and `--git-common-dir` for the caller; when they differ (caller is a linked worktree) it adds the common gitdir to the probe/index-scan set, deduplicated against the caller and every linked admin dir. Added two regressions that scan from a linked checkout and catch (a) a `secrets.json` staged in the main index and (b) a main `FETCH_HEAD` pinning a non-commit — both fail without the common-gitdir probe, pass with it.
